File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        if (root == null){
            return 0;
        }
        
        int resultLeft = 1;
        int resultRight = 1;

        if (root.left != null){
            resultLeft += maxDepth(root.left);
        }
        if (root.right != null){
            resultRight +=maxDepth(root.right);
        }

        if(resultLeft>resultRight){
            return resultLeft;
        }
        else{
            return resultRight;
        }
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        maxSum = root.val
        
        def dfs(node):
            nonlocal maxSum

            if node == None:
                return 0
            
            maxLeft = max(dfs(node.left), 0)
            maxRight = max(dfs(node.right), 0)

            maxSum = max(node.val + maxLeft + maxRight, maxSum)
            
            return node.val + max(maxLeft, maxRight)
        
        dfs(root)
        return maxSum
                

